### ISSUE TITLE
[Typo] 09.Templates_I.pdf

### ISSUE THREAD

**Issue by jakeheke75** [federico-busato/Modern-CPP-Programming#41]
Hi Federico,
Not sure if I have eaten too much, I have a couple of doubts at page 15:
- Return type of the function `print1` should be `void`
- Default values must be specified at the end of the parameters list only for primary class templates, primary variable template or alias template as explained [here](https://en.cppreference.com/w/cpp/language/template_parameters). Since here we have a function template there is no compile error for instruction `template<int A = 3, int B>`
![image](https://github.com/federico-busato/Modern-CPP-Programming/assets/113254017/dc1e07eb-adef-4c80-9d00-423af82bf712)
Thanks,
Gianmario

**Comment by jakeheke75** [comment]
Adding another small typo at page 23: in the comment below, it should be `// print 1, 2.0`
![image](https://github.com/federico-busato/Modern-CPP-Programming/assets/113254017/c4574442-2ebb-4034-a6bb-015c2601421e)
Small typo at page 24:  small typo missing opening round bracket
![image](https://github.com/federico-busato/Modern-CPP-Programming/assets/113254017/84dcc927-1197-432b-8a21-f85670fd55ac)
I have some doubts at page 27:
![image](https://github.com/federico-busato/Modern-CPP-Programming/assets/113254017/e97c4f3d-8139-4cf2-b17d-fe64810a2c73)
I tried to compile the code but I got some errors, not sure what I need to fix, some proposals:
- maybe function template `apply1` should return the result of function call `f` instead of `g`
- function template apply2 should have `int` return type and return the result of function call `f` instead of `g`
- definition of function `f` should come at the the top before template definition

I tried to build the following code and it was successful:
```cpp
int f(int a, int b) {
return a + b;
}

template<int (*)(int, int)> // <-- signature of "f"
int apply1(int a, int b) {
return f(a, b);
}

template<decltype(f)> // alternative syntax
int apply2(int a, int b) {
return f(a, b);
}

int main() {
    apply1<f>(2, 3); // return 5
    apply2<f>(2, 3); // return 5
} 
```
Thanks,
Gianmario

**Comment by federico-busato** [comment]
Hi Gianmario,
> Not sure if I have eaten too much

You are not alone 😂

As for the first problem, I was not aware that this was possible. My intuition was that it should be a compile error. I improved the slide this way 
![image](https://github.com/federico-busato/Modern-CPP-Programming/assets/55920584/449767c6-7cb7-45cc-897d-880c0b906177)

Fixed slide 24. While regarding slide 27, the idea is to call the function from the template. I updated the slide in this way  
![image](https://github.com/federico-busato/Modern-CPP-Programming/assets/55920584/2c1b83f6-38b7-445e-9849-d432d946d88e)
 
thanks again,
Federico
